api-admin/src/main/java/bio/terra/pearl/api/admin/service/forms/SurveyExtService.java

```diff
@@ -58,6 +64,59 @@ public Survey create(String portalShortcode, Survey survey, AdminUser adminUser)
     return surveyService.create(survey);
   }
 
+  public void delete(String portalShortcode, String surveyStableId, AdminUser adminUser) {
+    Portal portal = authUtilService.authUserToPortal(adminUser, portalShortcode);
+    List<Survey> existingVersions = surveyService.findByStableId(surveyStableId);
+    List<UUID> existingVersionIds = existingVersions.stream().map(BaseEntity::getId).toList();
+
+    if (existingVersionIds.size() == 0) {
+      throw new NotFoundException("Survey not found in portal " + portal.getName());
+    }
+
+    // Find all of the configured surveys that use any version of the specified survey
+    List<StudyEnvironmentSurvey> referencingSurveys =
+        existingVersionIds.stream()
+            .flatMap(
+                surveyId -> {
+                  List<StudyEnvironmentSurvey> configuredSurveys =
+                      studyEnvironmentSurveyService.findAllStudyEnvsWithSurveyId(surveyId);
+                  return configuredSurveys.stream();
+                })
+            .toList();
+
+    // Resolve all of the study environments that contain the configured surveys
+    List<StudyEnvironment> referencingStudyEnvs =
+        referencingSurveys.stream()
+            .map(
+                configuredSurvey -> {
+                  return studyEnvironmentService
+                      .find(configuredSurvey.getStudyEnvironmentId())
+                      .get();
+                })
+            .toList();
```

Comment:
Instead of getting the environments one-by-one, make a list of all the studyEnv ids, then grab them with findAll()
```suggestion
    List<StudyEnvironment> referencingStudyEnvs =
       studyEnvironmentService.findAll(
            referencingSurveys.stream()
                  .map( configuredSurvey -> StudyEnvironmentSurvey::getStudyEnvironmentId).toList()
       );
```